src/triggers.c

```diff
@@ -938,14 +938,28 @@ void hurt_touch()
 {
 	if (!other->s.v.takedamage)
 	{
+		if (cvar("k_ctf_hurt_items"))
+		{
+			if (streq(other->classname, "item_flag_team1") || streq(other->classname, "item_flag_team2"))
+			{
+				other->super_time = g_globalvars.time;
+			}
+			else if (streq(other->classname, "rune"))
+			{
+				other->s.v.nextthink = g_globalvars.time;
+			}
+		}
 		return;
 	}
 
-	self->s.v.solid = SOLID_NOT;
 	other->deathtype = dtTRIGGER_HURT;
 	T_Damage(other, self, self, self->dmg);
-	self->think = (func_t) hurt_on;
-	self->s.v.nextthink = g_globalvars.time + 1;
+	if (!ISDEAD(other))
+	{
+		self->s.v.solid = SOLID_NOT;
+		self->think = (func_t) hurt_on;
+		self->s.v.nextthink = g_globalvars.time + 1;
+	}
```

Comment:
This seems to be a possible workaround for the above mentioned problem with flag not getting destroyed, and generally a good idea as there's no point in considering periodic damage if player is already dead.